build/files.c

```diff
@@ -877,13 +886,29 @@ static VFA_t const virtualAttrs[] = {
  */
 static rpmRC parseForSimple(char * buf, FileEntry cur, ARGV_t * fileNames)
 {
-    char *s, *t;
+    char *s, *t, *end;
+    char *delim = " \t\n";
+    int quotes = 0;
     rpmRC res = RPMRC_OK;
     int allow_relative = (RPMFILE_PUBKEY|RPMFILE_DOC|RPMFILE_LICENSE);
 
     t = buf;
-    while ((s = strtokWithQuotes(t, " \t\n")) != NULL) {
```

Comment:
Not introduced or changed in this patch, but on a related note: it's more than a little hysterical that we split the line with tabs and spaces, forcing all this quoting on them, BUT then we don't allow more than one file per line anyway. Except for special %doc and %license.

So unless I'm missing something here, we could split by newline alone when allow_relative is false, and have spaces and tabs in filenames "just work" with zero quoting.

Reply:
That's a good point and I don't think you're missing anything. In fact, we might as well just simplify the whole quotation semantics as follows:
- No quotes means interpret globs as usual.
- Quotes mean don't interpret any globs.
That way, if we split on newlines, spaces wouldn't have to be dealt with in any special way. We also wouldn't need to support backslash escaping *at all* since quotes would do that. This would simplify #1903 quite significantly.

The current approach I chose for #1903 is to support backslashes properly, also support them for spaces, and to keep the possibility to allow multiple files per line by removing the "artificial" conditional we have for it, however taking a step back now, it does seem to be a little bit on the overkill side of the spectrum :smile:

Reply:
Hmm, I thought I was commenting in the shell-globbing PR because those commits are in here. That's the intended context for the above though.

Reply:
Yup, I figured, my reply above is also intended for the context of that PR.